tools/datafix/delete_bugs_with_source.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+"""Utility to delete all bugs for a given source."""
+
+from google.cloud import datastore
+from google.cloud.datastore.query import PropertyFilter
+
+import argparse
+
+MAX_BATCH_SIZE = 500
+
+
+def main() -> None:
+  parser = argparse.ArgumentParser(
+      description="Delete all bugs from a given source.")
+  parser.add_argument(
+      "--dry-run",
+      action=argparse.BooleanOptionalAction,
+      dest="dryrun",
+      default=True,
+      help="Abort before making changes")
+  parser.add_argument(
+      "--verbose",
+      action=argparse.BooleanOptionalAction,
+      dest="verbose",
+      default=False,
+      help="Display records being operated on")
+  parser.add_argument(
+      "--source",
+      action="store",
+      dest="source",
+      default="cve-osv",
+      help="The prefix of source_id records to delete")
+  parser.add_argument(
+      "--project",
+      action="store",
+      dest="project",
+      default="oss-vdb-test",
+      help="GCP project to operate on")
+  args = parser.parse_args()
+
+  client = datastore.Client(project=args.project)
+
+  query = client.query(kind="Bug")
```

Comment:
This can run a lot faster as a keys_only query when verbose is set to false.

Reply:
But if it's already running at a decent speed no need to optimise it.

Reply:
If I'm reading https://cloud.google.com/datastore/docs/samples/datastore-keys-only-query#datastore_keys_only_query-python correctly, it seems like a pretty straightforward optimisation? Thank you for bringing this to my attention.

Reply:
Added in [16860f6](https://github.com/google/osv.dev/pull/1660/commits/16860f6df6bedd3b02ddf4cccb17aa0ec125e195)